src/OSPSuite.Core/Domain/Services/FormulaTask.cs

```diff
@@ -298,10 +298,11 @@ private void updateLumenNavigationSegmentReferencingPath(EntityFormulaPath entit
 
       private IContainer getContainerOrThrow(IReadOnlyList<string> path, IEntity entity)
       {
+         var entityPath = _entityPathResolver.FullPathFor(entity);
```

Comment:
Resolve only in the if before throwing. Why do you get it all the time ?

Reply:
yes.